biotracks/createdp.py

```diff
@@ -24,80 +24,73 @@
 # POSSIBILITY OF SUCH DAMAGE.
 # #L%
 
+import json
 import csv
-import io
 import os
 import re
 
-import datapackage as dp
+import datapackage
 from jsontableschema import infer
 from . import cmso
+from .utils import get_logger, mkdir_p
 
 
 # https://specs.frictionlessdata.io/data-package/#metadata
 NAME_PATTERN = re.compile(r"^[a-z0-9_.-]+$")
+FOREIGN_KEYS = [{
+    "fields": cmso.OBJECT_ID,
+    "reference": {
+        "datapackage": "",
+        "resource": cmso.OBJECTS_TABLE,
+        "fields": cmso.OBJECT_ID
+    }
+}]
 
 
-def create_dpkg(top_level_dict, dict_, directory, joint_id):
-    """Create the datapackage representation.
+def infer_from_df(df, **kwargs):
```

Comment:
If I get this right, the '`infer_from_df`' substitutes the '`create_dpkg`'.
What does this function return? Maybe we should give it a more sensible name?